### PR TITLE
Fix max_data_size_for_stats with mask

### LINKED ISSUE #3207: max_data_size_for_stats returns incorrect results for encoded inputs

### PR DESCRIPTION
**Pull request by mbasmanova**
max_data_size_for_stats aggregated used to incorrectly process masked input. 

Fixes #3207

### CODE REVIEW

**Comment by mbasmanova** [comment]
CC: @shiyu-bytedance

File: velox/functions/prestosql/aggregates/MaxSizeForStatsAggregate.cpp [hunk 1]
```diff
@@ -123,40 +123,35 @@ class MaxSizeForStatsAggregate
...
-    }
-
     if (decoded_.isConstantMapping()) {
-      estimateSerializedSizes(arg, rows, 1);
-      rows.applyToSelected([&](vector_size_t i) {
-        updateOneAccumulator(groups[i], elementSizes_, 0);
-      });
+      if (!decoded_.isNullAt(0)) {
+        estimateSerializedSizes(arg, rows, 1);
+        rows.applyToSelected([&](vector_size_t i) {
```
Comment (xiaoxmeng) [inline velox/functions/prestosql/aggregates/MaxSizeForStatsAggregate.cpp:146]:
```
nit: s/i/row/
```

File: velox/functions/prestosql/aggregates/MaxSizeForStatsAggregate.cpp [hunk 2]
```diff
@@ -191,21 +186,18 @@ class MaxSizeForStatsAggregate
...
+        updateOneAccumulator(group, 0, elementSizes_[0]);
       }
-      // Estimate first element because it is constant mapping.
-      estimateSerializedSizes(arg, rows, 1);
-      updateOneAccumulator(group, elementSizes_, 0);
-      return;
+    } else {
+      estimateSerializedSizes(arg, rows, rows.countSelected());
+      vector_size_t sizeIndex = 0;
+      rows.applyToSelected([&](vector_size_t i) {
```
Comment (xiaoxmeng) [inline velox/functions/prestosql/aggregates/MaxSizeForStatsAggregate.cpp:197]: ditto

File: velox/functions/prestosql/aggregates/MaxSizeForStatsAggregate.cpp [hunk 3]
```diff
@@ -191,21 +186,18 @@ class MaxSizeForStatsAggregate
     decoded_.decode(*arg, rows, true);
 
     if (decoded_.isConstantMapping()) {
-      if (decoded_.isNullAt(0)) {
-        // There's nothing to do; all values are null.
-        return;
+      if (!decoded_.isNullAt(0)) {
+        // Estimate first element because it is constant mapping.
+        estimateSerializedSizes(arg, rows, 1);
```
Comment (xiaoxmeng) [inline velox/functions/prestosql/aggregates/MaxSizeForStatsAggregate.cpp:191]: ditto

File: velox/functions/prestosql/aggregates/MaxSizeForStatsAggregate.cpp [hunk 4]
```diff
@@ -123,40 +123,35 @@ class MaxSizeForStatsAggregate
...
+      if (!decoded_.isNullAt(0)) {
+        estimateSerializedSizes(arg, rows, 1);
+        rows.applyToSelected([&](vector_size_t i) {
+          updateOneAccumulator(groups[i], i, elementSizes_[0]);
+        });
+      }
     } else {
       estimateSerializedSizes(arg, rows, rows.countSelected());
       vector_size_t sizeIndex = 0;
       rows.applyToSelected([&](vector_size_t i) {
```
Comment (xiaoxmeng) [inline velox/functions/prestosql/aggregates/MaxSizeForStatsAggregate.cpp:153]: ditto

**Review by xiaoxmeng — APPROVED** [review]
@mbasmanova LGTM. Thanks for the fix!

**Pull request closed without merging**